syntax = "proto3";

service WeightSync{
  // stream request
  rpc SynchronizeWeights(stream SynchronizeWeightsRequest) returns(SynchronizeWeightsResponse){}
}

message NpArray {
  bytes array = 1;
}

message SynchronizeWeightsRequest{
  repeated string keys = 1;
  repeated NpArray weights = 2;
}

message SynchronizeWeightsResponse{
  string result = 1;
}


